pyfakefs/helpers.py

```diff
@@ -426,3 +435,42 @@ def getvalue(self) -> bytes:
 
     def putvalue(self, value: bytes) -> None:
         self._bytestream.write(value)
+
+
+def is_called_from_skipped_module(skip_names: list) -> bool:
+    stack = traceback.extract_stack()
+    from_open_code = (
+        sys.version_info >= (3, 12)
+        and stack[0].name == "open_code"
+        and stack[0].line == "return self._io_module.open_code(path)"
+    )
+
+    caller_filename = next(
+        (
+            frame.filename
+            for frame in stack[::-1]
+            if not frame.filename.startswith("<frozen ")
+            and not frame.filename.startswith(STDLIB_PATH)
+            and (
+                not frame.filename.startswith(PYFAKEFS_PATH)
+                or any(
+                    frame.filename.startswith(test_path)
+                    for test_path in PYFAKEFS_TEST_PATHS
+                )
+            )
+        ),
```

Comment:
This looks like a good idea to me. I'm currently working on Python 3.13 compatibility, and the previous approach with the hardcoded stackframe size already does not work there, due to the `pathlib` module now implemented as a directory with the implementing modules inside (which also may call each other). It may decrease the performance a bit, but I hope that it will not be much.
I haven't used `sysconfig.get_path("stdlib")` before, but it looks like the correct way (it does handle global, user and virtual environments, right?).